Library/EnvelopeFactory.ts

```diff
@@ -1,13 +1,163 @@
+import os = require("os");
 import Contracts = require("../Declarations/Contracts")
 import Util = require("./Util")
 import Config = require("./Config");
 import Context = require("./Context");
 import { CorrelationContextManager } from "../AutoCollection/CorrelationContextManager";
 
+var StreamId = Util.w3cTraceId(); // Create a guid
+
 /**
  * Manages the logic of creating envelopes from Telemetry objects
  */
 class EnvelopeFactory {
+    public static createQuickPulseEnvelope(metrics: Contracts.MetricQuickPulse[], documents: Contracts.DocumentQuickPulse[], config: Config, context: Context): Contracts.EnvelopeQuickPulse {
+        const machineName = os.hostname(); // Note: os.hostname() was added in node v0.3.3
+        const instance = (context.tags
+            && context.keys
+            && context.keys.cloudRoleInstance
+            && context.tags[context.keys.cloudRoleInstance]) || machineName;
+
+        var envelope: Contracts.EnvelopeQuickPulse = {
+            Documents: documents.length > 0 ? documents : null,
+            InstrumentationKey: config.instrumentationKey || "",
+            Metrics: metrics.length > 0 ? metrics : null,
+            InvariantVersion: 1, //  1 -> v1 QPS protocol
+            Timestamp: `\/Date(${Date.now()})\/`,
+            Version: context.tags[context.keys.internalSdkVersion],
+            StreamId: StreamId,
+            MachineName: machineName,
+            Instance: instance
+        }
+
+        return envelope;
+    }
+
+    public static createQuickPulseMetric(
+        telemetry: Contracts.MetricTelemetry
+    ): Contracts.MetricQuickPulse {
+        var data: Contracts.MetricQuickPulse;
+        data = {
+            Name: telemetry.name, // TODO: map from MetricTelemetry name to QuickPulse name
+            Value: telemetry.value,
+            Weight: telemetry.count || 1
+        };
+        return data;
+    }
+
+    public static telemetryEnvelopeToQuickPulseDocument(envelope: Contracts.Envelope): Contracts.DocumentQuickPulse {
+        switch (envelope.data.baseType) {
+            case "ExceptionData":
+                return EnvelopeFactory.createQuickPulseExceptionDocument(envelope);
+            case "MessageData":
+                return EnvelopeFactory.createQuickPulseTraceDocument(envelope);
+            case "RemoteDependencyData":
+                return EnvelopeFactory.createQuickPulseDependencyDocument(envelope);
+            case "RequestData":
+                return EnvelopeFactory.createQuickPulseRequestDocument(envelope);
+        }
+        return null;
+    }
+
+    private static createQuickPulseTraceDocument(envelope: Contracts.Envelope): Contracts.MessageDocumentQuickPulse {
+        const document = EnvelopeFactory.createQuickPulseDocument(envelope);
+
+        var traceDocument: Contracts.MessageDocumentQuickPulse = {
+            ...document,
+            Properties: [{key: 'my key', value: 'my value'}],
```

Comment:
Are these supposed to be here?

Reply:
Added properties/measurements aggregation